### PR TITLE
Support request

### PR DESCRIPTION
**Pull request by sashank007**

### CODE REVIEW

File: frontend/src/Components/Main/SupportAreas/SupportAreas.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React, { Component } from "react";
+
+class SupportAreas extends React.Component {
```
Comment (rbruce2) [inline frontend/src/Components/Main/SupportAreas/SupportAreas.js]: What is this for?

File: frontend/src/Components/Main/SupportRequest/SupportRequest.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from "react";
+import HorizontalStepper from "../HorizontalStepper/HorizontalStepper";
 
 class SupportRequest extends React.Component {
   render() {
     return (
       <div>
-        <h1>Support Request</h1>
+        {/* <h1>Support Request</h1> */}
```
Comment (rbruce2) [inline frontend/src/Components/Main/SupportRequest/SupportRequest.js]: Cleanup comments

File: routes/teamworkDesk.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+var express = require("express"),
+  router = express.Router(),
+  db = require("../models"),
+  Authentication = require("../middleware/Authentication"),
+  config = require("../middleware/config"),
+  cas = Authentication.getInstance(config);
+const axios = require("axios");
+var apiToken = `0ueH80iKQ5M8duyll58kxqtVPWHJKUt2srbanEgyyL4M2UtcoM`;
```
Comment (rbruce2) [inline routes/teamworkDesk.js]: Remove and add as env variable... We never upload api keys to github

File: routes/teamworkDesk.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      return response;
+    })
+    .catch(error => {
+      console.log("error -->", data, error.response);
+      return error.response;
+    });
+});
+
+//upload form end point,
+router.post("/fileUpload", upload.single("file"), (req, res) => {
```
Comment (rbruce2) [inline routes/teamworkDesk.js]: Lets try to handle everything with axios.

File: frontend/src/Components/Main/HorizontalStepper/HorizontalStepper.js [hunk 5]
```diff
@@ -0,0 +1,592 @@
...
+        checkMark: false,
+        path:
+          "M18.65,2.85L19.26,6.71L22.77,8.5L21,12L22.78,15.5L19.24,17.29L18.63,21.15L14.74,20.54L11.97,23.3L9.19,20.5L5.33,21.14L4.71,17.25L1.22,15.47L3,11.97L1.23,8.5L4.74,6.69L5.35,2.86L9.22,3.5L12,0.69L14.77,3.46L18.65,2.85M9.5,7A1.5,1.5 0 0,0 8,8.5A1.5,1.5 0 0,0 9.5,10A1.5,1.5 0 0,0 11,8.5A1.5,1.5 0 0,0 9.5,7M14.5,14A1.5,1.5 0 0,0 13,15.5A1.5,1.5 0 0,0 14.5,17A1.5,1.5 0 0,0 16,15.5A1.5,1.5 0 0,0 14.5,14M8.41,17L17,8.41L15.59,7L7,15.59L8.41,17Z"
+      }
+    ],
+    subArea: null,
+    location: "",
+    fileUrl: [],
+    attachmentId: [],
+    userDetails: {
```
Comment (rbruce2) [inline frontend/src/Components/Main/HorizontalStepper/HorizontalStepper.js]: Using multiple user states?

File: frontend/src/Components/Main/HorizontalStepper/HorizontalStepper.js [hunk 6]
```diff
@@ -0,0 +1,592 @@
...
+          <div>
+            {activeStep === steps.length ? (
+              <div>
+                <Typography>
+                  <h2 className="instructions">
+                    Your ticket has been received!
+                  </h2>
+                  <p>
+                    A member of our team will be in contact as soon as possible.
+                    We'll reply to tickets in the order they were received.
```
Comment (rbruce2) [inline frontend/src/Components/Main/HorizontalStepper/HorizontalStepper.js]: Handle reset?

File: frontend/src/Components/Main/FileDroppa/FileDroppa.js [hunk 7]
```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import MagicDropzone from "react-magic-dropzone";
+import { render } from "react-dom";
+import FadeSnackBar from "../FadeSnackBar/FadeSnackBar";
+import "./FileDroppa.css";
+const notAValidFileMessage = "This file type is not permitted.";
+
+class FileDroppa extends React.Component {
```
[the remark this answers is not in the document]
Comment (sashank007) [reply frontend/src/Components/Main/FileDroppa/FileDroppa.js]: Renamed.

File: frontend/src/Components/Main/SupportAreas/SupportAreas.js (hunk 1 above)
Comment (sashank007) [reply frontend/src/Components/Main/SupportAreas/SupportAreas.js]: Initially used this to create SupportAreas form. Now using AreaForm. Deleted this.

File: frontend/src/Components/Main/SupportRequest/SupportRequest.js (hunk 2 above)
Comment (sashank007) [reply frontend/src/Components/Main/SupportRequest/SupportRequest.js]: Done.

File: frontend/src/Components/Main/HorizontalStepper/DetailsForm/DetailsForm.css [hunk 8]
```diff
@@ -0,0 +1,77 @@
+/* #root
```
[the remark this answers is not in the document]
Comment (sashank007) [reply frontend/src/Components/Main/HorizontalStepper/DetailsForm/DetailsForm.css]: Done

File: frontend/src/Components/Main/FileDroppa/FileDroppa.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    previews.splice(i, 1);
+    selectedFiles.splice(i, 1);
+    console.log("new previews", previews);
+    this.setState({ previews, selectedFiles });
+  };
+
+  render() {
+    return (
+      <div className="Dropzone-page">
+        {/* <div className="Dropzone-input-wrapper"> */}
```
[the remark this answers is not in the document]
Comment (sashank007) [reply frontend/src/Components/Main/FileDroppa/FileDroppa.js]: Done

File: frontend/src/Components/Main/HorizontalStepper/DetailsForm/DetailsForm.js [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+
+    return {
+      emailAddress: emailAddress,
+      name: displayName,
+      department: primaryDepartment,
+      firstName: firstName,
+      lastName: lastName
+    };
+  }
+  // fetchData() {
```
[the remark this answers is not in the document]
Comment (sashank007) [reply frontend/src/Components/Main/HorizontalStepper/DetailsForm/DetailsForm.js]: Done

File: routes/teamworkDesk.js (hunk 3 above)
Comment (sashank007) [reply routes/teamworkDesk.js]: Done

**Comment by sashank007** [comment]
Ready for review.

**Pull request merged**